### PR TITLE
WhatsApp notification example

### PR DESCRIPTION
**Pull request by pushnitsa**

### CODE REVIEW

**Comment by pushnitsa** [comment]
> Error	NuGet Package restore failed for project VirtoCommerce.WhatsAppNotification.Web: Unable to find version '2.13.48' of package 'VirtoCommerce.Platform.Core'.
> C:\Program Files (x86)\Microsoft SDKs\NuGetPackages: Package 'VirtoCommerce.Platform.Core.2.13.48' is not found on source 'C:\Program Files (x86)\Microsoft SDKs\NuGetPackages'.
> https://api.nuget.org/v3/index.json: Package 'VirtoCommerce.Platform.Core.2.13.48' is not found on source 'https://api.nuget.org/v3/index.json'.
> 
> **Can you reference a released version 2.13.47?**

Done. Please Check.

File: WhatsAppNotification/VirtoCommerce.WhatsAppNotification.Web/Module.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+                    Body = ReadFile($"{typeof(OrderWhatsAppNotification).Name}_body.htm")
+                }
+            });
+
+            var eventHandlerRegistrar = _container.Resolve<IHandlerRegistrar>();
+
+            eventHandlerRegistrar.RegisterHandler<OrderChangedEvent>(async (message, token) => await _container.Resolve<SendNotificationOrderChangedEventHandler>().Handle(message));
+        }
+
+        private static string ReadFile(string fileName)
```
Comment (eMazeika) [inline WhatsAppNotification/VirtoCommerce.WhatsAppNotification.Web/Module.cs]: remove this method. Refactor to use .ReadToString() extension from namespace VirtoCommerce.Platform.Core.Common

File: WhatsAppNotification/VirtoCommerce.WhatsAppNotification.Web/module.manifest [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+<module>
+  <id>VirtoCommerce.WhatsAppNotification</id>
+  <version>1.0.2</version>
+
+  <platformVersion>2.13.47</platformVersion>
+  <dependencies>
+  </dependencies>
+
+  <title>WhatsApp notification module</title>
+  <description>This module provide WhatsApp notification gateway</description>
```
Comment (eMazeika) [inline WhatsAppNotification/VirtoCommerce.WhatsAppNotification.Web/module.manifest]: Sample sending Order change notifications to WhatsApp via Twilio or Woowa gateways

File: WhatsAppNotification/VirtoCommerce.WhatsAppNotification.Data/Gateway/WhatsAppNotificationSendingGateway.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
+using System;
+using VirtoCommerce.Platform.Core.Notifications;
+using VirtoCommerce.WhatsAppNotification.Core.Gateways;
+using VirtoCommerce.WhatsAppNotification.Data.Senders;
+
+namespace VirtoCommerce.WhatsAppNotification.Data.Gateway
```
Comment (eMazeika) [inline WhatsAppNotification/VirtoCommerce.WhatsAppNotification.Data/Gateway/WhatsAppNotificationSendingGateway.cs]: rename folder/namespace .Data/gateway to .Data/Gateways

File: WhatsAppNotification/VirtoCommerce.WhatsAppNotification.Data/Handlers/SendNotificationOrderChangedEventHandler.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+                {
+                    var notification = _notificationManager.GetNewNotification<OrderWhatsAppNotification>();
+
+                    var user = await _securityService.FindByIdAsync(changedEntry.OldEntry.CustomerId, UserDetails.Reduced);
+
+                    var contact = user != null
+                        ? _memberService.GetByIds(new[] { user.MemberId }).FirstOrDefault()
+                        : _memberService.GetByIds(new[] { changedEntry.OldEntry.CustomerId }).FirstOrDefault();
+
+                    notification.Recipient = contact?.Phones?.FirstOrDefault(x => !string.IsNullOrEmpty(x)) ?? user?.PhoneNumber;
```
[the remark this answers is not in the document]
Comment (eMazeika) [reply WhatsAppNotification/VirtoCommerce.WhatsAppNotification.Data/Handlers/SendNotificationOrderChangedEventHandler.cs]: validate the `PhoneNumber` first, then create `notification` only inside the `if` statement, not earlier

**Pull request merged**